src/util/ConfigManager/ConfigManager.h

```diff
@@ -60,47 +62,84 @@ class ConfigManager {
   @param variableToPutValueOfTheOptionIn The value held by the configuration
   option will be copied into this variable, whenever the value in the
   configuration option changes.
-  @param defaultValue A default value for the configuration option. If none is
-  given, signified by an empty optional, then a value for the configuration
-  option MUST be given at runtime.
+
+  @return A pointer to the newly created configuration option.
   */
   template <typename OptionType>
   requires ad_utility::isTypeContainedIn<OptionType,
                                          ConfigOption::AvailableTypes>
-  void createConfigOption(const std::vector<std::string>& pathToOption,
-                          std::string_view optionDescription,
-                          OptionType* variableToPutValueOfTheOptionIn,
-                          std::optional<OptionType> defaultValue =
-                              std::optional<OptionType>(std::nullopt)) {
-    /*
-    We need a non-empty path to construct a ConfigOption object, the `verify...`
-    function always throws an exception for this case. No need to duplicate the
-    error code.
-    */
-    if (pathToOption.empty()) {
-      verifyPathToConfigOption(pathToOption, "");
-    }
+  const ConfigOption* addOption(const std::vector<std::string>& pathToOption,
```

Comment:
All of those should return `const ConfigOption&`, we don't have any use for a pointer.
And also state in the documentation, that the return value stays valid even after adding other options.

Reply:
Done, see [207ea5e](https://github.com/ad-freiburg/qlever/pull/1040/commits/207ea5e832e793d2576687fec685d05d46cea289).